### PR TITLE
Fix parallel testing

### PR DESCRIPTION
**Pull request by astrofrog**
Fixes https://github.com/astropy/astropy/issues/2871

Also trying to see if I can run the parallel tests on CircleCI

cc @drdavella

### CODE REVIEW

**Comment by bsipocz** [comment]
@pllim - why does this have a windows tag?

**Comment by pllim** [comment]
@astrofrog says "CircleCI" so that was why, but feel free to remove tag if that was wrong.

**Comment by bsipocz** [comment]
circleCI runs a 32bit linux box :)

**Comment by astrofrog** [comment]
Important note: enabling parallel containers in CircleCI is not useful, as it runs 4 separate containers, making it hard to test this. So for now I've disabled it (but we can still test the parallel option, there just won't be any speedup)

**Comment by pllim** [comment]
Oh... I thought it's 32-bit Windows, not sure why! My bad. Thanks for fixing the labels, @bsipocz !

**Comment by bsipocz** [comment]
> Important note: enabling parallel containers in CircleCI is not useful, as it runs 4 separate containers, making it hard to test this. So for now I've disabled it (but we can still test the parallel option, there just won't be any speedup)

What about having then only 2 parallel threads, also maybe just for one module? If there is no speed up, then running 4 doesn't make any sense.

**Comment by astrofrog** [comment]
> What about having then only 2 parallel threads, also maybe just for one module? If there is no speed up, then running 4 doesn't make any sense

There is no harm in running on 4 though, the runtime is not any worse. Since we can't do a build matrix on CircleCI, I think just running the full CI with 4 threads is fine?

**Comment by astrofrog** [comment]
@eteq - just for info, this fixes a minor 'bug' with Python 2.7 in coordinates. It turns out that:

```
python setup.py test -t docs/time/index.rst
```

actually raises an exception because if run on its own, ``_site_registry`` from astropy.coordinates is not yet defined, and it tries to get downloaded. This doesn't happen in Python 3.6 because the remote_data exception is a URLError there (but it's an IOError in Python 2). Anyway, details, but bottom line is that the above test command now works with Python 2 with this PR.

**Comment by astrofrog** [comment]
@bsipocz @eteq - do you fancy reviewing this?

**Comment by bsipocz** [comment]
Looks good to me!

File: astropy/coordinates/earth.py [hunk 1]
```diff
@@ -476,7 +476,11 @@ def _get_site_registry(cls, force_download=False, force_builtin=False):
                         reg = get_downloaded_sites(force_download)
                     else:
                         reg = get_downloaded_sites()
-                except six.moves.urllib.error.URLError:
+                except (six.moves.urllib.error.URLError, IOError):
```
Comment (bsipocz) [inline astropy/coordinates/earth.py]: nice catch!

**Comment by bsipocz** [comment]
Since this still has some issues on travis I went ahead and merged #6420 that contained only the version limitation for circleCI. Now all the other PRs can be rebased.

**Pull request merged**